orchestrator/services/processes.py

```diff
@@ -158,31 +159,30 @@ def _update_process(process_id: UUID, step: Step, process_state: WFProcess) -> P
 def _get_current_step_to_update(
     stat: ProcessStat, p: ProcessTable, step: Step, process_state: WFProcess
 ) -> ProcessStepTable:
-    """Checks if last error state is identical to determine if we add a new step or update the last one."""
+    """Checks if last error state is identical to determine if we add a new step or update the last one.
+
+    There are some special internal keys checked to customise how the step is updated in the database.
+    These keys are only used for this purpose and not stored in the database.
+    - '__step_name_override' - When this key is present, the value will be used as the ProcessStepTable.name
+    - '__replace_last_state' - When the value for this key is truthy, the previous state in the db will be overridden
+    - '__remove_keys' - Removes the keys in this given list from the state
```

Comment:
Maybe it's an idea to group all of these keys in the state, in a key called '__runtime' or whatever, and then shape it through a pydantic model?

```python
class RuntimeVariables(BaseModel):
    step_name_override: str | None = Field(None, description="When this key is present, the value will be used as ProcessStepTable.name")
    replace_last_state: bool | None = Field(None, description="...")
    remove_keys: list[str] | None = Field(None, description="...")


# serialize
variables = RuntimeVariables(step_name_override="my_step", remove_keys=["__step_group"])
step_state['__runtime'] = variables.dict() 


# deserialize
runtime = RuntimeVariables(**step_state.get("__runtime", {}))
```

I think it can save a bit of type checking, validation and dict manipulation spreading through the codebase